ldp/nn/graph/llm_call_op.py

```diff
@@ -92,8 +92,12 @@ def prep_tools_for_tokenizer(tools: list[Tool] | None) -> list[dict] | None:
             return None
 
         # TODO: should be able to switch to tool.info.model_dump() here
-        return [
-            {
+        tools_list = []
+        for tool in tools:
+            if tool.info.parameters is None:
+                raise NotImplementedError("Cannot serialize tools with no parameters.")
```

Comment:
```suggestion
                raise NotImplementedError("Didn't yet handle serializing tools without parameters.")
```

It's not that we can't, it's that we didn't add this yet